### PR TITLE
[PWGCF] Generic Framework: PID v2

### PR DESCRIPTION
**Pull request by Preet-Bhanjan**
Using generic framework for computing the two-particle correlation v2 for pions, kaons, and protons.

### CODE REVIEW

**Review by victor-gonzalez — CHANGES_REQUESTED** [review]
Please, don't use the `GenericFramework` directory. It is for the own `GFW`

This task was already created in the Flow directory which is where it should be placed

Why is not the one there updated?

**Comment by Preet-Bhanjan** [comment]
This task uses the GFW and so the GenericFramework directory was being used.

However, since the same task is in the Flow directory, I will modify it to
use the GFW.

File: PWGCF/Flow/Tasks/FlowPbPbpikp.cxx [hunk 1]
```diff
@@ -8,494 +8,297 @@
...
-#include "Framework/AnalysisTask.h"
-#include "Framework/HistogramRegistry.h"
-#include "Framework/StepTHn.h"
-#include "Framework/runDataProcessing.h"
-#include "PWGLF/DataModel/LFStrangenessTables.h"
+
+#include "PWGCF/GenericFramework/Core/GFWPowerArray.h"
+#include "PWGCF/GenericFramework/Core/GFW.h"
+#include "PWGCF/GenericFramework/Core/GFWCumulant.h"
+#include "FlowContainer.h"
```
Comment (victor-gonzalez) [inline PWGCF/Flow/Tasks/FlowPbPbpikp.cxx:40]: Either use the full path for the `FlowContainer.h` include or remove it for the other components
It is misleading suggesting that they are in different places

File: PWGCF/Flow/Tasks/FlowPbPbpikp.cxx [hunk 2]
```diff
@@ -8,494 +8,297 @@
...
 using namespace o2;
 using namespace o2::framework;
 using namespace o2::framework::expressions;
-using std::array;
 
-struct v2ellip {
+#define O2_DEFINE_CONFIGURABLE(NAME, TYPE, DEFAULT, HELP) Configurable<TYPE> NAME{#NAME, DEFAULT, HELP};
 
-  // Connect to ccdb
+struct testgf {
```
Comment (victor-gonzalez) [inline PWGCF/Flow/Tasks/FlowPbPbpikp.cxx:54]: Please, chose a better name

File: PWGCF/Flow/Tasks/FlowPbPbpikp.cxx [hunk 3]
```diff
@@ -8,494 +8,297 @@
...
+    corrconfigs.push_back(fGFW->GetCorrelatorConfig("poiNpi refN08 | olNpi {2} refP08 {-2}", "Pi08Gap22", kTRUE));
+    corrconfigs.push_back(fGFW->GetCorrelatorConfig("poiNk refN08 | olNk {2} refP08 {-2}", "Ka08Gap22", kTRUE));
+    corrconfigs.push_back(fGFW->GetCorrelatorConfig("poiNpr refN08 | olNpr {2} refP08 {-2}", "Pr08Gap22", kTRUE));
+    fGFW->CreateRegions();
   }
 
-  template <typename T>
-  bool selectionTrack(const T& candidate)
+  template <typename TTrack>
+  std::pair<int, int> GivebayesID(TTrack track)
```
Comment (victor-gonzalez) [inline PWGCF/Flow/Tasks/FlowPbPbpikp.cxx:162]: `GivebayesID` -> `GetBayesId`?

File: PWGCF/Flow/Tasks/FlowPbPbpikp.cxx [hunk 4]
```diff
@@ -8,494 +8,297 @@
...
+    int maxProb[3] = {80, 80, 80};
+    int pidID = -1;
+    std::pair<int, int> idprob = GivebayesID(track);
+    if (idprob.first == 0 || idprob.first == 1 || idprob.first == 2) { // 0 = pion, 1 = kaon, 2 = proton
+      pidID = idprob.first;
+      float nsigmaTPC[3] = {track.tpcNSigmaPi(), track.tpcNSigmaKa(), track.tpcNSigmaPr()};
+      if (idprob.second > maxProb[pidID]) {
+        if (abs(nsigmaTPC[pidID]) > 3)
+          return 0;
+        return pidID + 1; // shift the pid by 1, 1 = pion, 2 = kaon, 3 = proton
```
Comment (victor-gonzalez) [inline PWGCF/Flow/Tasks/FlowPbPbpikp.cxx:189]: I would suggest using an `enum` for particle named constants and using it consistently
The function can return `-1` (`kWRONGID = -1`) in case the identification is not accepted

File: PWGCF/GenericFramework/Tasks/CMakeLists.txt [hunk 5]
```diff
@@ -12,4 +12,4 @@
 o2physics_add_dpl_workflow(flow-generic-framework
```
Comment (victor-gonzalez) [inline PWGCF/GenericFramework/Tasks/CMakeLists.txt:12]: This file should not be part of the PR

**Review by victor-gonzalez — CHANGES_REQUESTED** [review]
Please, remove from the PR the not affected file and also change the name of the task according to the functionality
The other comments are for your consideration in next iterations

File: PWGCF/GenericFramework/Tasks/CMakeLists.txt (hunk 5 above)
Comment (victor-gonzalez) [inline PWGCF/GenericFramework/Tasks/CMakeLists.txt:12]: This file should not be part of this PR

**Review by victor-gonzalez — CHANGES_REQUESTED** [review]
The file `PWGCF/GenericFramework/Tasks/CMakeLists.txt` should not be part of this PR which means that it should not be modified by the PR. As it is right now the PR modify it and that's why it is included in the PR. It should not be included in the PR

**Comment by victor-gonzalez** [comment]
@Preet-Bhanjan  Please, don't close the PR, all the comments will be lost if you open a new one
Just modify your local repository restoring the original `PWGCF/GenericFramework/Tasks/CMakeLists.txt` before your changes and then push again towards your remote fork. That will update automatically the PR

**Review by victor-gonzalez — APPROVED** [review]
Thanks!
